tests/test_oauth2_provider.py

```diff
@@ -303,3 +304,36 @@ def test_token_endpoint_unauth(self):
         resp = self.provider.token_endpoint(request=areq.to_urlencoded())
         atr = TokenErrorResponse().deserialize(resp.message, "json")
         assert _eq(atr.keys(), ['error_description', 'error'])
+
+
+    def test_response_types(self):
```

Comment:
I would split this into two tests, with a common setup/helper method that initializes self.provider.cdb.

Basically both:
1) create an authreq
2) call auth_init() with it
3) assert something

Reply:
Made something along those lines.